src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -175,8 +171,13 @@ BoundExpression MakeIsDeclarationPattern(SyntaxNode syntax, BoundExpression lowe
             }
             else if (type.IsValueType)
             {
+                // The type here is not a Nullable<T> instance type, as that would have led to the semantic error:
+                // ERR_PatternNullableType: It is not legal to use nullable type '{0}' in a pattern; use the underlying type '{1}' instead.
+                Debug.Assert(!type.IsNullableType());
+
                 // It is possible that the input value is already of the correct type, in which case the pattern
-                // is irrefutable, and we can just do the assignment and return true.
+                // is irrefutable (there is no way for a non-nullable value type to be null), and we can just do
+                // the assignment and return true.
                 if (loweredInput.Type == type)
```

Comment:
That may be true (the suggested change would be a code improvement), but it is not relevant to the change in the PR.